src/reanimated2/WorkletEventHandler.ts

```diff
@@ -20,64 +23,121 @@ function jsListener<Event extends object>(
   };
 }
 
-export default class WorkletEventHandler<Event extends object> {
+class WorkletEventHandlerNative<Event extends object>
+  implements IWorkletEventHandler<Event>
+{
+  eventNames: string[];
   worklet: (event: ReanimatedEvent<Event>) => void;
+  #viewTags: Set<number>;
+  #registrations: Map<number, number[]>; // keys are viewTags, values are arrays of registration ID's for each viewTag
+  constructor(
+    worklet: (event: ReanimatedEvent<Event>) => void,
+    eventNames: string[] = []
+  ) {
+    this.worklet = worklet;
+    this.eventNames = eventNames;
+    this.#viewTags = new Set<number>();
+    this.#registrations = new Map<number, number[]>();
+  }
+
+  updateEventHandler(
+    newWorklet: (event: ReanimatedEvent<Event>) => void,
+    newEvents: string[]
+  ): void {
+    // Update worklet and event names
+    this.worklet = newWorklet;
+    this.eventNames = newEvents;
+
+    // Detach all events
+    this.#registrations.forEach((registrationIDs, _tag) => {
+      registrationIDs.forEach((id) => {
+        unregisterEventHandler(id);
+      });
+      // No need to remove registrationIDs from map, since it gets overwritten when attaching
+    });
+
+    // Attach new events with new worklet
+    Array.from(this.#viewTags).forEach((tag) => {
+      const newRegistrations = this.eventNames.map((eventName) => {
+        return registerEventHandler(this.worklet, eventName, tag);
+      });
```

Comment:
We can skip `return` here I think
```suggestion
      const newRegistrations = this.eventNames.map((eventName) => registerEventHandler(this.worklet, eventName, tag));
```